sites/public/src/components/listing/FilterDrawerHelper.tsx

```diff
@@ -0,0 +1,313 @@
+import {
+  EnumListingFilterParamsComparison,
+  FilterAvailabilityEnum,
+  HomeTypeEnum,
+  ListingFeatures,
+  ListingFilterKeys,
+  ListingFilterParams,
+  RegionEnum,
+  UnitTypeEnum,
+} from "@bloom-housing/shared-helpers/src/types/backend-swagger"
+import styles from "./FilterDrawer.module.scss"
+import { UseFormMethods } from "react-hook-form"
+import { Grid } from "@bloom-housing/ui-seeds"
+import { Field, t } from "@bloom-housing/ui-components"
+import { isTrue } from "../../lib/helpers"
+import { encode, ParsedUrlQuery } from "querystring"
+
+export interface FilterData {
+  availability?: Record<FilterAvailabilityEnum, boolean | "true" | "false">
+  bedroomTypes?: Record<UnitTypeEnum, boolean | "true" | "false">
+  homeType?: Record<HomeTypeEnum, boolean | "true" | "false">
+  isVerified?: boolean | "true" | "false"
+  listingFeatures?: Record<keyof ListingFeatures, boolean | "true" | "false">
+  monthlyRent?: Record<"maxRent" | "minRent", string>
+  regions?: Record<RegionEnum, boolean | "true" | "false">
+  section8Acceptance?: boolean | "true" | "false"
+}
+
+export interface FilterField {
+  key: string
+  label: string
+  defaultChecked: boolean
+}
+
+export interface CheckboxGroupProps {
+  groupLabel: string
+  fields: FilterField[]
+  register: UseFormMethods["register"]
+  customRowNumber?: number
+}
+
+export interface RentSectionProps {
+  register: UseFormMethods["register"]
+  getValues: UseFormMethods["getValues"]
+  setValue: UseFormMethods["setValue"]
+  filterState: FilterData
+}
+
+const arrayFilters: ListingFilterKeys[] = [
+  ListingFilterKeys.bedroomTypes,
+  ListingFilterKeys.counties,
+  ListingFilterKeys.homeTypes,
+  ListingFilterKeys.listingFeatures,
+  ListingFilterKeys.regions,
+  ListingFilterKeys.reservedCommunityTypes,
+  ListingFilterKeys.availabilities,
+]
+
+const booleanFilters: ListingFilterKeys[] = [
+  ListingFilterKeys.isVerified,
+  ListingFilterKeys.section8Acceptance,
+]
+
+// two filters below have yet to be implemented, captured in part 4
+const indvidualFilters: ListingFilterKeys[] = [
+  ListingFilterKeys.bathrooms,
+  ListingFilterKeys.jurisdiction,
+]
+
+export const unitTypeMapping = {
+  [UnitTypeEnum.studio]: 0,
+  [UnitTypeEnum.oneBdrm]: 1,
+  [UnitTypeEnum.twoBdrm]: 2,
+  [UnitTypeEnum.threeBdrm]: 3,
+  [UnitTypeEnum.fourBdrm]: 4,
+  [UnitTypeEnum.fiveBdrm]: 5,
+}
+
+export const buildDefaultFilterFields = (
+  filterType: ListingFilterKeys,
+  stringBase: string,
+  keyArr: string[],
+  existingData: FilterData
+): FilterField[] =>
+  keyArr.map((key) => {
```

Comment:
issue: This might be covered in a different ticket, but noticed that these aren't getting translated. Same goes for "Home Type"